lib/pat.c

```diff
@@ -5990,6 +5990,12 @@ pat_key_defrag_each(grn_ctx *ctx,
   return new_curr_key;
 }
 
+static inline void
+pat_clear_garbages(grn_pat *pat)
+{
+  memset(pat->header->garbages, 0, sizeof(grn_id) * (GRN_PAT_MAX_KEY_SIZE + 1));
```

Comment:
Although I don't have the confident about it, how about clearing pat->header->garbages in the same way to create new pat because it could be efficient? 

https://github.com/groonga/groonga/blob/4fc5d9cd349d8c3f1f65ad4dd4b35888d7f9b89f/lib/pat.c#L1548

Reply:
コメントをもらったときから処理を変えたので、`n_garbages` も `0` で初期化するようにしました。